Source/AwsCommonRuntimeKit/http/HttpMessage.swift

```diff
@@ -29,54 +28,53 @@ public class HttpMessage {
         self.rawValue = rawValue
     }
 
-    init(headers: HttpHeaders, allocator: Allocator = defaultAllocator) throws {
-        self.allocator = allocator
-        guard let rawValue = aws_http_message_new_request_with_headers(allocator.rawValue, headers.rawValue) else {
-            throw CommonRunTimeError.crtError(.makeFromLastError())
-        }
-        self.rawValue = rawValue
-    }
-
     deinit {
         aws_http_message_release(rawValue)
     }
 }
-// header handling
+
 public extension HttpMessage {
 
     var headerCount: Int {
-           return aws_http_message_get_header_count(rawValue)
+        return aws_http_message_get_header_count(rawValue)
     }
 
-    // Todo: what to do in error? Maybe refactor it to a better logic?
-    func addHeaders(headers: HttpHeaders) {
-        for index in 0...headers.count {
-            if let header = headers.get(index: index) {
-                aws_http_message_add_header(self.rawValue, header)
-            }
+    func addHeader(header: HttpHeader) throws {
+        guard (header.withCStruct { cHeader in
+            aws_http_message_add_header(self.rawValue, cHeader)
+        }) == AWS_OP_SUCCESS else {
+            throw CommonRunTimeError.crtError(.makeFromLastError())
         }
-       // self.headers = headers
     }
 
-    func removeHeader(atIndex index: Int) -> Bool {
-        return aws_http_message_erase_header(rawValue, index) == AWS_OP_SUCCESS
+    func addHeaders(headers: [HttpHeader]) throws {
+        try headers.forEach { try addHeader(header: $0) }
     }
 
-    func getHeader(atIndex index: Int) -> HttpHeader? {
+    /// Get header at index. Index must be valid.
+    func removeHeader(atIndex index: Int) {
+        guard aws_http_message_erase_header(rawValue, index) == AWS_OP_SUCCESS else {
+            fatalError("Index out of range")
+        }
+    }
+
+    /// Get header at index. Index must be valid.
+    func getHeader(atIndex index: Int) -> HttpHeader {
         var header = aws_http_header()
-        if aws_http_message_get_header(self.rawValue, &header, index) == AWS_OP_SUCCESS {
-            return HttpHeader(rawValue: header)
+        guard aws_http_message_get_header(self.rawValue, &header, index) == AWS_OP_SUCCESS else {
+            fatalError("Index out of range")
         }
-        return nil
+        return HttpHeader(rawValue: header)
     }
```

Comment:
Crash if index is invalid. This is same as Swift's API https://developer.apple.com/documentation/swift/array/remove(at:)-1p2pj